lib/globus_client.rb

```diff
@@ -56,19 +56,27 @@ def disallow_writes(...)
     endpoint.disallow_writes
   end
 
+  # NOTE: Can't use the `...` (argument forwarding) operator here because we
+  #       want to route the keyword args to `Endpoint#new` and the block arg to
+  #       `Endpoint#list_files`
+  def list_files(**keywords, &block)
+    endpoint = Endpoint.new(config, **keywords)
+    endpoint.list_files(&block)
+  end
+
   def file_count(...)
     endpoint = Endpoint.new(config, ...)
-    endpoint.file_count
+    endpoint.list_files { |files| return files.count }
   end
 
   def total_size(...)
     endpoint = Endpoint.new(config, ...)
-    endpoint.total_size
+    endpoint.list_files { |files| return files.sum(&:size) }
   end
 
   def get_filenames(...)
     endpoint = Endpoint.new(config, ...)
-    endpoint.get_filenames
+    endpoint.list_files { |files| return files.map(&:name) }
```

Comment:
I decided not to remove these methods from the public API and instead dog-food a bit. I could be convinced that they no longer need to be in the public API, though.